### PR TITLE
Ts migration: final

### PR DESCRIPTION
**Pull request by Tristan-WorkGH**
Require to merge PRs:
  - #506
  - #511

---

This PR do:
  - migrate last JS files to TS and remove `allowJs` from tsconfig
  - fix cause of `skipLibCheck`:
    * add missing `@types` for _**json-logic-js** (dep of commons-ui/react-querybuilder)_ and _**react-virtualized** (dep of commons-ui)_
    * fix module augmentation of mui and emotion theme definition

### CODE REVIEW

File: src/module-emotion.d.ts [hunk 1]
```diff
@@ -8,11 +8,10 @@
...
 
-import { Theme as EmotionTheme } from '@emotion/react';
+// import { Theme as EmotionTheme } from '@emotion/react';
 import { Theme as MaterialUITheme } from '@mui/material';
 
-//import '@emotion/react';
 declare module '@emotion/react' {
     // Re-declare the emotion theme to have the properties of the MaterialUiTheme
-    export interface Theme extends EmotionTheme, MaterialUITheme {}
+    export interface Theme extends /*EmotionTheme,*/ MaterialUITheme {}
```
Comment (TheMaskedTurtle) [inline src/module-emotion.d.ts:16]: why ?

File: src/module-mui.d.ts [hunk 2]
```diff
@@ -25,9 +24,9 @@ declare module '@mui/material/styles/createTheme' {
         };
         agGridBackground: CSSObject;
     };
-    export interface Theme extends MuiTheme, ThemeExtension {}
+    export interface Theme extends /*MuiTheme,*/ ThemeExtension {}
     // allow configuration using `createTheme`
-    export interface ThemeOptions extends MuiThemeOptions, Partial<ThemeExtension> {}
+    export interface ThemeOptions extends /*MuiThemeOptions,*/ Partial<ThemeExtension> {}
```
Comment (TheMaskedTurtle) [inline src/module-mui.d.ts:29]: why ?

Comment (Tristan-WorkGH) [reply src/module-mui.d.ts:29]: <pre>&gt; tsc &amp;&amp; vite build

<font color="#34E2E2">src/module-mui.d.ts</font>:<font color="#FCE94F">27</font>:<font color="#FCE94F">22</font> - <font color="#EF2929">error</font><font color="#555753"> TS2310: </font>Type &apos;Theme&apos; recursively references itself as a base type.

<span style="background-color:#FFFFFF"><font color="#300A24">27</font></span>     export interface Theme extends MuiTheme, ThemeExtension {}
<span style="background-color:#FFFFFF"><font color="#300A24">  </font></span> <font color="#EF2929">                     ~~~~~</font>

<font color="#34E2E2">src/module-mui.d.ts</font>:<font color="#FCE94F">29</font>:<font color="#FCE94F">22</font> - <font color="#EF2929">error</font><font color="#555753"> TS2310: </font>Type &apos;ThemeOptions&apos; recursively references itself as a base type.

<span style="background-color:#FFFFFF"><font color="#300A24">29</font></span>     export interface ThemeOptions extends MuiThemeOptions, Partial&lt;ThemeExtension&gt; {}
<span style="background-color:#FFFFFF"><font color="#300A24">  </font></span> <font color="#EF2929">                     ~~~~~~~~~~~~</font>


Found 2 errors in the same file, starting at: src/module-mui.d.ts<font color="#555753">:27</font>
</pre>

File: src/module-emotion.d.ts (hunk 1 above)
Comment (Tristan-WorkGH) [reply src/module-emotion.d.ts:16]: <pre>&gt; gridexplore-app@0.1.0 build
&gt; tsc &amp;&amp; vite build

<font color="#34E2E2">src/module-emotion.d.ts</font>:<font color="#FCE94F">16</font>:<font color="#FCE94F">22</font> - <font color="#EF2929">error</font><font color="#555753"> TS2310: </font>Type &apos;Theme&apos; recursively references itself as a base type.

<span style="background-color:#FFFFFF"><font color="#300A24">16</font></span>     export interface Theme extends EmotionTheme, MaterialUITheme {}
<span style="background-color:#FFFFFF"><font color="#300A24">  </font></span> <font color="#EF2929">                     ~~~~~</font>


Found 1 error in src/module-emotion.d.ts<font color="#555753">:16</font>
</pre>

Comment (Meklo) [reply src/module-emotion.d.ts:16]: Since we're augmenting emotion's theme here maybe it wasn't relevant to reinject it into itself in the first place

**Review by Meklo — APPROVED** [review]

**Review by TheMaskedTurtle — APPROVED** [review]

**Pull request merged**